backend/api/quivr_api/modules/sync/utils/sync.py

```diff
@@ -0,0 +1,606 @@
+                    credentials["access_token"],
+                    oauth2_refresh_token=credentials["refresh_token"],
+                    app_key=os.getenv("DROPBOX_APP_KEY"),
+                    oauth2_access_token_expiration=credentials.get("expires_at"),
+                    app_secret=os.getenv("DROPBOX_APP_SECRET"),
+                )
+            self.dbx.check_and_refresh_access_token()
+            credentials["access_token"] = self.dbx._oauth2_access_token
+
+            def fetch_files(metadata):
+                files = []
+                for file in metadata.entries:
+
+                    shared_link = f"https://www.dropbox.com/preview{file.path_display}?context=content_suggestions&role=personal"
+                    is_folder = isinstance(file, dropbox.files.FolderMetadata)
+
+                    files.append(
+                        SyncFile(
+                            name=file.name,
+                            id=file.id,
+                            is_folder=is_folder,
+                            last_modified=(
+                                str(file.client_modified) if not is_folder else ""
+                            ),
+                            mime_type=(
+                                file.path_lower.split(".")[-1] if not is_folder else ""
+                            ),
+                            web_view_link=shared_link,
+                        )
+                    )
+                return files
+
+            files = []
+            list_metadata = self.dbx.files_list_folder(folder_id, recursive=recursive)
+            files.extend(fetch_files(list_metadata))
+
+            while list_metadata.has_more:
+                list_metadata = self.dbx.files_list_folder_continue(
+                    list_metadata.cursor
+                )
+                files.extend(fetch_files(list_metadata))
+
+            for file in files:
+                file.name = remove_special_characters(file.name)
+
+            logger.info("Dropbox files retrieved successfully: %d", len(files))
+            return files
+
+        except dropbox.exceptions.ApiError as e:
+            logger.error("Dropbox API error: %s", e)
+            return []
+        except Exception as e:
+            logger.error("Unexpected error: %s", e)
+            return []
+
+    def get_files_by_id(
+        self, credentials: Dict[str, str], file_ids: List[str]
+    ) -> List[SyncFile]:
+        """
+        Retrieve files from Dropbox by their IDs.
+
+        Args:
+            credentials (dict): The credentials for accessing Dropbox.
+            file_ids (list): The list of file IDs to retrieve.
+
+        Returns:
+            list: A list of dictionaries containing the metadata of each file or an error message.
+        """
+        logger.info("Retrieving Dropbox files with file_ids: %s", file_ids)
+
+        if "access_token" not in credentials:
+            logger.error("Access token is not in the credentials")
```

Comment:
If this function has to fail maybe we shouldn't return a []. The caller has to check the empty list and raise. If this is expected behavior then never mind.